### DISCUSSION TITLE
[design decision] Optional use of double quotes around keys and strings

### DISCUSSION THREAD

**Discussion by Fiurilli** [tabular-json/tabular-json#6] [Ideas]
Interesting concept for sure, JSON files are easy to work with but can sometimes be quite large so the inclusion a csv like format is intriguing. I understand the desire to make quotes around keys and strings optional, but I think that makes the files very prone to user error. 

For example in the playground the string _First Avenue_ does not require the double quotes, but the string _1st Avenue_ does since it starts with an integer. When modifying a file manually it is easy to make mistakes such as these, even for experienced users. I would therefore make it as fool proof as possible by enforcing double quotes around all keys and strings, just as the JSON file format does.

**Comment by josdejong** [comment]
Thanks for your inputs!

This is indeed a trade-off: for the human eye, data is so much more clean and readable without all the quotes. It's one of the things I like about CSV. But it introduces relatively complex rules about when quotes are required. For parsers, ease of understanding, and hence overall robustness, it's easier to have mandatory quotes.

Good point, let's rethink optional quotes.

**Comment by josdejong** [comment]
I've been thinking about this and discussed with multiple people. Tabular-JSON is in the first place a data format (and not a configuration format). Therefore, unambiguity and ease of parsing is very important. When having unquoted strings, it is quite hard to explain when quotes are needed exactly and when not. This is an issue with CSV too, and maybe we shouldn't inherit that.

Though I personally would love to have optional quotes, I think the simplicity of always requiring quotes weighs heavier than looking cleaner for the human eye when having optional quotes. Therefore I'll remove support for optional quotes from Tabular-JSON via #9.

**Comment by josdejong** [comment]
Just came across an interesting article about the consequences of optional quotes 😅: [The Norway Problem](https://hitchdev.com/strictyaml/why/implicit-typing-removed/)
